x-pack/plugins/alerting/server/alerts_client/lib/get_persistent_alerts_query.ts

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  QueryDslQueryContainer,
+  SearchRequest,
+  SearchTotalHits,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { merge } from 'lodash';
+import {
+  ALERT_END,
+  ALERT_INSTANCE_ID,
+  ALERT_MAINTENANCE_WINDOW_IDS,
+  ALERT_RULE_EXECUTION_UUID,
+  ALERT_RULE_UUID,
+  ALERT_START,
+  EVENT_ACTION,
+  TIMESTAMP,
+} from '@kbn/rule-data-utils';
+import { Alert } from '@kbn/alerts-as-data-utils';
+import { AlertsFilter, ISO_WEEKDAYS, RuleAlertData } from '../../../common';
+import {
+  GetLifecycleAlertsQueryByExecutionUuidParams,
+  GetLifecycleAlertsQueryByTimeRangeParams,
+  GetAlertsQueryParams,
+  GetQueryByExecutionUuidParams,
+  GetQueryByTimeRangeParams,
+  SearchResult,
+} from '../types';
+import { PersistentAlertsChunk } from '../..';
+import { FormatAlert } from '../../types';
+
+const MAX_ALERT_DOCS_TO_RETURN = 100;
+enum AlertTypes {
+  NEW = 0,
+  ONGOING,
+  RECOVERED,
+}
+
+const getLifecycleAlertsQueryByExecutionUuid = ({
```

Comment:
can we add tests for these functions to ensure the shape of the query is as expected?

Reply:
Queries are tested for each type of combinations:
LifeCycle, Continual,
getExpectedQueryByExecutionUuid, getExpectedQueryByTimeRange,
new, ongoing, recovered,

in the alertsClient unit test.

here: https://github.com/elastic/kibana/pull/161061/files#diff-deec7c2dd986076158ce26636e8d1282ab66e824ed1cf4ef7ee873a797446bc9R1278